util/stopper.go

```diff
@@ -50,8 +50,8 @@ type Stopper struct {
 	stopper  chan struct{}  // Closed when stopping
 	stopped  chan struct{}  // Closed when stopped completely
 	stop     sync.WaitGroup // Incremented for outstanding workers
-	mu       sync.Mutex     // Protects the fields below
 	draining int32          // 1 when Stop() has been called, updated atomically
```

Comment:
please pardon my ignorance: why not use a bool here that is always updated under lock?